src/registries/npm-registry.js

```diff
@@ -23,6 +23,12 @@ const DEFAULT_REGISTRY = 'https://registry.npmjs.org/';
 const REGEX_REGISTRY_PREFIX = /^https?:/;
 const REGEX_REGISTRY_SUFFIX = /registry\/?$/;
 
+// All scoped package names are of the format `@scope%2fpkg` from the use of NpmRegistry.escapeName
+// `(?:^|\/)` Match either the start of the string or a `/` but don't capture
+// `[^\/?]*?` Match any character that is not '/' or '?' and capture, up until the first occurance of:
+// `%2f` Match '%2f' the escaped forward slash and don't capture
+const SCOPED_PKG_REGEXP = /(?:^|\/)(@[^\/?]*?)(?=%2f)/;
```

Comment:
This matches an empty scope like `@%2f`. Also, matches `@scope%2f`without the package name. Gonna merge anyways since I don't think this is a problem but I'm quite confused about the `*?` part in `(@[^\/?]*?)`. Why not just `(@[^\/?]+)` instead. What does even `*?` mean since both of them are quantifiers and you're not supposed to use two quantifiers back to back.

_Goes away, reads about `*?` making `*` not greedy._

I'm now convinced that we need `+` instead of `*?` since our aim is to match a scope and the full name of the scope. So gonna replace that and then merge it. LMK if you disagree.